packages/backend/src/models/ProjectModel/ProjectModel.ts

```diff
@@ -407,15 +409,16 @@ export class ProjectModel {
     async saveExploresToCache(
         projectUuid: string,
         explores: (Explore | ExploreError)[],
-    ): Promise<{ projectUuid: string; explores: Explore[] }> {
-        return this.database(CachedExploresTableName)
+    ): Promise<DbCachedExplores> {
+        const [cachedExplores] = await this.database(CachedExploresTableName)
             .insert({
                 project_uuid: projectUuid,
                 explores: JSON.stringify(explores),
             })
             .onConflict('project_uuid')
             .merge()
-            .select(['project_uuid', 'explores']);
+            .returning('*');
+        return cachedExplores;
```

Comment:
`cachedExplores` can also be `undefined` in the case that no results are returned from the query.

Reply:
mm, I've been copying this code from other inserts, we never check if the result is undefined. is this supposed to happen on `insert`s ?

Reply:
eg: https://github.com/lightdash/lightdash/blob/main/packages/backend/src/models/ProjectModel/ProjectModel.ts#L185

Reply:
Ah but the insert is probably always successful. I'm not certain but I would guess it always inserts

Reply:
If it doesn't insert it should throw an error but I agree with you for the selects. 

> Typescript doesn't raise errors for indexing, which is annoying
We can search for eslint rules that might help us with this. We had bugs across the app before because of this.